test/itkTorchImageTest.cxx

```diff
@@ -66,203 +133,160 @@ int itkTorchImageTest(int argc, char *argv[])
-    ImageType::Pointer image = ImageType::New();
-  }
-  {
-    using ImageType = itk::TorchImage< double, 1 >;
-    ImageType::Pointer image = ImageType::New();
-  }
-  {
-    using ImageType = itk::TorchImage< itk::RGBPixel< short >, 3 >;
-    ImageType::Pointer image = ImageType::New();
-  }
-  {
-    using ImageType = itk::TorchImage< itk::RGBAPixel< short >, 1 >;
-    ImageType::Pointer image = ImageType::New();
+    using PixelType = unsigned char;
+    constexpr int ImageDimension = 3;
+    const std::string StructName = "TorchImage<unsigned char, 3>";
+    const int SizePerDimension = 16;
+    const PixelType firstValue = 10;
+    const PixelType secondValue = 130;
+    const PixelType thirdValue = 12;
+    itkTorchImageTestByTypeAndDimension< PixelType, ImageDimension >( SizePerDimension, StructName, firstValue, secondValue, thirdValue );
   }
   {
-    using ImageType = itk::TorchImage< itk::Vector< short, 3 >, 4 >;
-    ImageType::Pointer image = ImageType::New();
+    using PixelType = signed char;
+    constexpr int ImageDimension = 4;
+    const std::string StructName = "TorchImage<signed char, 4>";
+    const int SizePerDimension = 10;
+    const PixelType firstValue = 10;
+    const PixelType secondValue = -11;
+    const PixelType thirdValue = 12;
+    itkTorchImageTestByTypeAndDimension< PixelType, ImageDimension >( SizePerDimension, StructName, firstValue, secondValue, thirdValue );
   }
   {
-    using ImageType = itk::TorchImage< itk::CovariantVector< short, 4 >, 5 >;
-    ImageType::Pointer image = ImageType::New();
+    using PixelType = short;
+    constexpr int ImageDimension = 2;
+    const std::string StructName = "TorchImage<short, 2>";
+    const int SizePerDimension = 256;
+    const PixelType firstValue = 32000;
+    const PixelType secondValue = -32000;
+    const PixelType thirdValue = 5;
+    itkTorchImageTestByTypeAndDimension< PixelType, ImageDimension >( SizePerDimension, StructName, firstValue, secondValue, thirdValue );
   }
   {
-    using ImageType = itk::TorchImage< itk::Vector< itk::Vector< unsigned char, 2 >, 3 >, 4 >;
-    ImageType::Pointer image = ImageType::New();
+    using PixelType = int;
+    constexpr int ImageDimension = 5;
+    const std::string StructName = "TorchImage<int, 5>";
+    const int SizePerDimension = 8;
+    const PixelType firstValue = 2147483000;
+    const PixelType secondValue = -2147483000;
+    const PixelType thirdValue = 10;
+    itkTorchImageTestByTypeAndDimension< PixelType, ImageDimension >( SizePerDimension, StructName, firstValue, secondValue, thirdValue );
   }
   {
-    using ImageType = itk::TorchImage< itk::CovariantVector< itk::Vector< itk::RGBAPixel< unsigned char >, 2 >, 3 >, 4 >;
-    ImageType::Pointer image = ImageType::New();
+    using PixelType = long;
+    constexpr int ImageDimension = 1;
+    const std::string StructName = "TorchImage<long, 1>";
+    const int SizePerDimension = 1000;
+    const PixelType firstValue = 9223372036854775000L;
```

Comment:
`long` is 32-bit on Windows. You might want to switch this to `long long`.

Reply:
I will use `int16_t`, `int32_t`, and `int64_t` instead of `short`, `int`, and `long`, respectively, in this test function.